New resource ttl representation

michaelabenyohai opened the pull request:
All Views that extend RepresentationView now have to implement a turtleRepresentation() method, which returns a Jena Model object. When the NewTurtleWriter is called it obtains the Model for the View and writes to Turtle.

Commit:
Implement Turtle representation for each new resource

The NewTurtleWriter uses the turtleRepresentation from each RepresentationView to write the
RDF Model in TTL

Commit:
Remove new-data-formats html as we have now implemented all data formats

Commit:
Create tests for NewTurtleWriter

Also make sure ListValues are serialized correctly to TTL.
Also refactor to make sure we are creating links from RegisterDomainConfiguration rather than HttpRequest

philandstuff commented on src/main/java/uk/gov/register/presentation/representations/NewTurtleWriter.java:
```diff
@@ -0,0 +1,24 @@
+public class NewTurtleWriter extends NewRepresentationWriter {
+    
+    public NewTurtleWriter() {
+    }
```
very minor thing, but I don't think this constructor is needed

philandstuff commented on src/main/java/uk/gov/register/presentation/view/RegisterDetailView.java:
```diff
@@ -49,4 +50,9 @@ public RegisterDetail getRegisterDetail() {
+
+    @Override
+    public Model turtleRepresentation() {
+        throw new NotImplementedException("Not implented");
```
why are we implementing RepresentaionView if we don't want to actually implement it?

philandstuff commented on src/test/resources/fixtures/list.ttl:
```diff
@@ -1,11 +1,24 @@
+              <http://register.test.register.gov.uk/item/sha-256:877d8bd1ab71dc6e48f64b4ca83c6d7bf645a1eb56b34d50fa8a833e1101eb18> .
+
+<http://register.test.register.gov.uk/record/value2>
+      field:fields field:field1 , field:field2 ;
```
huh, I didn't know you could do lists in turtle like this

philandstuff commented:
looks good on first glance. I'll try to give it a more thorough review after lunch.

I'm a little concerned that this makes our view classes quite bloated (eg ItemView) with code that's specific only to RDF stuff. I'd be interested in having a conversation about we might avoid this (and whether it's worth it)

Commit:
Move logic to create Jena RDF Model from Views to individual TurtlerWriters

Also create CsvRepresentationView for which only views that have a
CSV representation are to extend

Commit:
Rename class

michaelabenyohai commented:
@philandstuff - latest commits should address your review comments.

philandstuff commented:
🐢
